tests/fixed_buf.rs

```diff
@@ -137,6 +139,54 @@ fn slicing() {
     })
 }
 
+#[test]
+fn pool_next_as_concurrency_limit() {
+    tokio_uring::start(async move {
+        const BUF_SIZE: usize = 80;
+
+        let mut tempfile = tempfile();
+        let file = StdFile::options()
+            .write(true)
+            .open(tempfile.path())
+            .unwrap();
+
+        let buffers = FixedBufPool::new(iter::repeat_with(|| Vec::with_capacity(BUF_SIZE)).take(2));
+        buffers.register().unwrap();
+
+        let mut join_handles = vec![];
+        for i in 0..10 {
+            let mut buf = buffers.next(BUF_SIZE).await;
+            println!(
+                "[main] iteration {}: obtained buffer {}",
+                i,
+                buf.buf_index()
+            );
+            let cloned_file = file.try_clone().unwrap();
+
+            let handle = tokio_uring::spawn(async move {
+                let file = File::from_std(cloned_file);
```

Comment:
There should be a way to share `File` and socket objects between tasks, since the descriptor is already in a `SharedFd` internally.